tools/slang-test/test-reporter.cpp

```diff
@@ -82,9 +84,10 @@ TestReporter::TestReporter() :
     m_isVerbose = false;
 }
 
-Result TestReporter::init(TestOutputMode outputMode)
+Result TestReporter::init(TestOutputMode outputMode, bool isSubReporter)
 {
     m_outputMode = outputMode;
+    m_isSubReporter = isSubReporter;
```

Comment:
This `m_isSubReporter` flag seems to be a sign that we need to step back and work on the architecture a bit. Can we separate out the functionality of the `TestReporter` into two different objects that support a 1-to-N relationship? Then the non-threaded cases is just the simple case when N is 1.